### PR TITLE
Publish container image

### LINKED ISSUE #104: Official container and kubernetes manifests associated with this project
Hi project maintainers,

Thanks for this helpful project. 
I'm wondering if there is any official container image and kubernetes manifests for this bond-cni?
I plan to use it with multus, I built the binady manually and tested it, worked as expected with multus.

Best regards!
Bassem

### PR DESCRIPTION
**Pull request by zeeke**
- Build plugin image
Add Dockerfile and related entrypoint.sh to build the binary and copy
it to the host.

- Build and publish images in CI
Add GitHub configuration to publish the docker image with the plugin binary

- Deployment file
With this configuration, `bond` CNI plugin can be deployed via:
```
kubectl apply -f https://raw.githubusercontent.com/k8snetworkplumbingwg/bond-cni/master/manifests/bond.yml
```

Fixes #104

### CODE REVIEW

**Comment by zeeke** [comment]
@mlguerrero12 @Eoghan1232 @SchSeba please take a look when you have time.

File: manifests/bond-cni.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        operator: Exists
+        effect: NoSchedule
+      containers:
+      - name: bond-cni-plugin
+        image: ghcr.io/k8snetworkplumbingwg/bond-cni:latest
+        resources:
+          requests:
+            cpu: "10m"
+            memory: "15Mi"
+        volumeMounts:
```
Comment (Eoghan1232) [inline manifests/bond-cni.yaml:36]: volume on host should not be "/host" ? 

did this work?  
ah I see in the entrypoint script you force create and copy this
I don't like this, we should reuse the already existing cni dir on the host?

**Review by Eoghan1232 — APPROVED** [review]
lgtm!

File: images/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# This Dockerfile is used to build the image available on DockerHub
+FROM golang:1.23 AS build
```
Comment (SchSeba) [inline images/Dockerfile:2]: nit: can you add `docker.io/`

File: images/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+WORKDIR /usr/src/bond-cni
+COPY . .
+RUN make build-bin
+
+FROM alpine:latest
```
Comment (SchSeba) [inline images/Dockerfile:8]: can you add `docker.io/`

File: images/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+LABEL org.opencontainers.image.source=https://github.com/k8snetworkplumbingwg/bond-cni
+WORKDIR /
+COPY --from=build /usr/src/bond-cni/bin .
+COPY --from=build /usr/src/bond-cni/LICENSE .
```
Comment (SchSeba) [inline images/Dockerfile:12]: nit: can we just copy directly and not from the other container?

File: images/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+COPY --from=build /usr/src/bond-cni/images/entrypoint.sh .
```
Comment (SchSeba) [inline images/Dockerfile:13]: nit: can we just copy directly and not from the other container?

File: manifests/bond-cni.yaml [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        - name: cnibin
+          mountPath: /host/opt/cni/bin/
+      volumes:
+        - name: cnibin
+          hostPath:
+            path: /host/opt/cni/bin/
```
Comment (SchSeba) [inline manifests/bond-cni.yaml:42]: nit: please add a new line here

Comment (SchSeba) [inline manifests/bond-cni.yaml:42]: This one doesn't see right the hostPath should be opt/cni/bin and inside the container we should have `/host/opt/cni/bin/` no?

**Review by SchSeba — APPROVED** [review]
LGTM

nice work!

**Pull request merged**